lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -538,6 +544,9 @@ def _on_opensearch_data_storage_detaching(self, _: StorageDetachingEvent):  # no
                 pass
         try:
             self._stop_opensearch()
+            if self.alt_hosts:
+                # There is enough peers available for us to try removing the unit
+                self.opensearch_exclusions.delete_current()
```

Comment:
We try to clean this unit out of the voting/allocation exclusions. The challenge: if we fail, its own `Scope.UNIT` entry that registers this exclusion will also go away. Therefore, we use the peer-departed to also register this departing unit to the app-level databag.